packages/ui-app/src/AddressSummary.tsx

```diff
@@ -261,20 +294,52 @@ class AddressSummary extends React.PureComponent<Props> {
       </Nonce>
     );
   }
+  protected renderSaveIcon () {
+    return <Button
+            className='saveButton'
+            onClick={this.saveName}
+            icon='save'
+            size='small'
+            isPrimary
+            key='save'
+    />;
+  }
 
-  protected renderChildren () {
-    const { children } = this.props;
+  protected renderUnlocking () {
+    const { accounts_idAndIndex = [], value, withUnlocking } = this.props;
+    const [_accountId] = accounts_idAndIndex;
+    const accountId = _accountId || value;
 
-    if (!children || (Array.isArray(children) && children.length === 0)) {
+    if (!withUnlocking || !accountId) {
       return null;
     }
 
     return (
-      <div className='ui--AddressSummary-children'>
-        {children}
-      </div>
+      <UnlockingDisplay
+        className='ui--AddressSummary-available'
+        params={accountId}
+      />
     );
   }
+
+  protected saveName = () => {
+    const { value } = this.props;
+    const { newName } = this.state;
+
+    const trimmedName = newName.trim();
+
+    // Save only if the name was changed or if it's no empty.
+    if (trimmedName !== '') {
+      const currentKeyring = value && keyring.getPair(value.toString());
```

Comment:
Would rather not enable the editor if not in the keyring. Here we are enabling, but then may not save it.

Reply:
Agreed, I guess having the keyring in the state wouldn't make sense as it doesn't change. I couldn't think of a better way than checking for `keyring.getPair(value.toString()` when first toggling the editor. lmk if there's a better ways.